plugins/auth-react/src/hooks/useCookieAuthRefresh/useCookieAuthRefresh.test.tsx

```diff
@@ -0,0 +1,287 @@
+    renderHook(() => useCookieAuthRefresh({ pluginId: 'techdocs' }), {
+      wrapper: ({ children }) => (
+        <TestApiProvider
+          apis={[
+            [fetchApiRef, fetchApiMock],
+            [discoveryApiRef, discoveryApiMock],
+          ]}
+        >
+          {children}
+        </TestApiProvider>
+      ),
+    });
+
+    await waitFor(() =>
+      expect(channelMock.addEventListener).toHaveBeenCalledTimes(1),
+    );
+
+    const twentyMinutesFromNowInMilliseconds =
+      now + 2 * tenMinutesInMilliseconds;
+
+    // simulating other tab refreshing the cookie
+    channelMock.postMessage({
+      action: 'COOKIE_REFRESHED',
+      payload: {
+        expiresAt: new Date(twentyMinutesFromNowInMilliseconds).toISOString(),
+      },
+    });
+
+    // advance the timers in 10 minutes to match the old expires at
+    jest.advanceTimersByTime(tenMinutesInMilliseconds);
+
+    // should not call the api
+    expect(fetchApiMock.fetch).toHaveBeenCalledTimes(1);
+
+    // advance the timers in more 10 minutes to match the new expires at
+    jest.advanceTimersByTime(tenMinutesInMilliseconds);
+
+    // should call the api
+    await waitFor(() => expect(fetchApiMock.fetch).toHaveBeenCalledTimes(2));
+  });
+
+  it('should cancel the refresh when the component is unmounted', async () => {
+    const { result, unmount } = renderHook(
+      () => useCookieAuthRefresh({ pluginId: 'techdocs' }),
+      {
+        wrapper: ({ children }) => (
+          <TestApiProvider
+            apis={[
+              [fetchApiRef, fetchApiMock],
+              [discoveryApiRef, discoveryApiMock],
+            ]}
+          >
+            {children}
+          </TestApiProvider>
+        ),
+      },
+    );
+
+    await waitFor(() => expect(fetchApiMock.fetch).toHaveBeenCalledTimes(1));
+
+    expect(result.current.state.result).toMatchObject({ expiresAt });
+
+    unmount();
+
+    expect(channelMock.removeEventListener).toHaveBeenCalledTimes(1);
+    expect(channelMock.removeEventListener).toHaveBeenCalledWith(
+      'message',
+      expect.any(Function),
+    );
+
+    // advance the timers to ensure that the refresh is not called
+    jest.advanceTimersByTime(tenMinutesInMilliseconds);
+
+    // should not call the api after unmount
+    await waitFor(() =>
+      expect(fetchApiMock.fetch).not.toHaveBeenCalledTimes(2),
+    );
+  });
+
+  it('should refresh the cookie when it is about to expire', async () => {
+    renderHook(() => useCookieAuthRefresh({ pluginId: 'techdocs' }), {
+      wrapper: ({ children }) => (
+        <TestApiProvider
+          apis={[
+            [fetchApiRef, fetchApiMock],
+            [discoveryApiRef, discoveryApiMock],
+          ]}
+        >
+          {children}
+        </TestApiProvider>
+      ),
+    });
+
+    await waitFor(() => expect(fetchApiMock.fetch).toHaveBeenCalledTimes(1));
+
+    // advance the timers to the expiration date
+    jest.advanceTimersByTime(tenMinutesInMilliseconds);
```

Comment:
can this need a margin of like a millisecond or so? :) seems to hit the EXACT expiry doesn't it

Reply:
Oh, you are right, we want to make sure the API is called before the expiration date 🙏🏻 , the test is fixed!